<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1183</width>
    <height>895</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="widget" native="true">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>28</y>
     <width>1121</width>
     <height>171</height>
    </rect>
   </property>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>内存块数：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>60</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>页面数量：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>19</x>
      <y>104</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>页号序列：</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="quick_table_check">
    <property name="geometry">
     <rect>
      <x>323</x>
      <y>143</y>
      <width>111</width>
      <height>22</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::RightToLeft</enum>
    </property>
    <property name="text">
     <string>启用快表</string>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLineEdit" name="memory_num_edit">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>16</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="taskSize_edit">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>55</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="task_edit">
    <property name="geometry">
     <rect>
      <x>119</x>
      <y>99</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="taskSize_btn">
    <property name="geometry">
     <rect>
      <x>327</x>
      <y>51</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>随机生成</string>
    </property>
   </widget>
   <widget class="QPushButton" name="task_btn">
    <property name="geometry">
     <rect>
      <x>326</x>
      <y>96</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>随机生成</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="memory_access_edit">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>16</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>613</x>
      <y>20</y>
      <width>131</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>内存存取时间：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>613</x>
      <y>100</y>
      <width>131</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>缺页中断时间：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>613</x>
      <y>60</y>
      <width>131</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>快表存取时间：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="miss_page_interrupt_edit">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>95</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="quick_table_access_edit">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>55</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="stop_btn">
    <property name="geometry">
     <rect>
      <x>964</x>
      <y>90</y>
      <width>112</width>
      <height>34</height>
     </rect>
    </property>
    <property name="text">
     <string>结束执行</string>
    </property>
   </widget>
   <widget class="QPushButton" name="stop_continue_btn">
    <property name="geometry">
     <rect>
      <x>964</x>
      <y>50</y>
      <width>112</width>
      <height>34</height>
     </rect>
    </property>
    <property name="text">
     <string>暂停执行</string>
    </property>
   </widget>
   <widget class="QPushButton" name="start_btn">
    <property name="geometry">
     <rect>
      <x>964</x>
      <y>10</y>
      <width>112</width>
      <height>34</height>
     </rect>
    </property>
    <property name="text">
     <string>开始执行</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="taskSize_low_edit">
    <property name="geometry">
     <rect>
      <x>435</x>
      <y>55</y>
      <width>41</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="taskSize_high_edit">
    <property name="geometry">
     <rect>
      <x>516</x>
      <y>55</y>
      <width>41</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_32">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>60</y>
      <width>16</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="task_high_edit">
    <property name="geometry">
     <rect>
      <x>516</x>
      <y>95</y>
      <width>41</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="task_low_edit">
    <property name="geometry">
     <rect>
      <x>435</x>
      <y>95</y>
      <width>41</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_33">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>100</y>
      <width>16</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_15">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>145</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>快表大小：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="quick_table_num_edit">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>140</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="run_time_edit">
    <property name="geometry">
     <rect>
      <x>751</x>
      <y>132</y>
      <width>181</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_16">
    <property name="geometry">
     <rect>
      <x>614</x>
      <y>137</y>
      <width>131</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>执行时间控制：</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_17">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>135</y>
      <width>21</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>ms</string>
    </property>
   </widget>
   <widget class="QPushButton" name="result_btn">
    <property name="geometry">
     <rect>
      <x>964</x>
      <y>130</y>
      <width>112</width>
      <height>34</height>
     </rect>
    </property>
    <property name="text">
     <string>立即推</string>
    </property>
   </widget>
   <zorder>label</zorder>
   <zorder>label_2</zorder>
   <zorder>label_3</zorder>
   <zorder>quick_table_check</zorder>
   <zorder>memory_num_edit</zorder>
   <zorder>taskSize_edit</zorder>
   <zorder>task_edit</zorder>
   <zorder>taskSize_btn</zorder>
   <zorder>task_btn</zorder>
   <zorder>memory_access_edit</zorder>
   <zorder>label_4</zorder>
   <zorder>label_5</zorder>
   <zorder>label_6</zorder>
   <zorder>miss_page_interrupt_edit</zorder>
   <zorder>quick_table_access_edit</zorder>
   <zorder>stop_continue_btn</zorder>
   <zorder>start_btn</zorder>
   <zorder>taskSize_low_edit</zorder>
   <zorder>taskSize_high_edit</zorder>
   <zorder>label_32</zorder>
   <zorder>task_high_edit</zorder>
   <zorder>task_low_edit</zorder>
   <zorder>label_33</zorder>
   <zorder>label_15</zorder>
   <zorder>quick_table_num_edit</zorder>
   <zorder>run_time_edit</zorder>
   <zorder>label_16</zorder>
   <zorder>label_17</zorder>
   <zorder>result_btn</zorder>
   <zorder>stop_btn</zorder>
  </widget>
  <widget class="QWidget" name="widget_2" native="true">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>760</y>
     <width>701</width>
     <height>60</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QPushButton" name="save_btn">
      <property name="text">
       <string>保存</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="open_btn">
      <property name="text">
       <string>打开</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="clear_btn">
      <property name="text">
       <string>清空</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="widget_5" native="true">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>550</y>
     <width>1101</width>
     <height>201</height>
    </rect>
   </property>
   <widget class="QLabel" name="label_11">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>13</y>
      <width>72</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>中断次数</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_12">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>59</y>
      <width>54</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>总时间</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>105</y>
      <width>72</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>平均时间</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>13</x>
      <y>151</y>
      <width>54</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>缺页率</string>
    </property>
   </widget>
   <widget class="QLabel" name="fifo_interrupt_label">
    <property name="geometry">
     <rect>
      <x>175</x>
      <y>13</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lru_interrupt_label">
    <property name="geometry">
     <rect>
      <x>417</x>
      <y>13</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lfu_interrupt_label">
    <property name="geometry">
     <rect>
      <x>676</x>
      <y>13</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="opt_interrupt_label">
    <property name="geometry">
     <rect>
      <x>947</x>
      <y>13</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="fifo_total_time_label">
    <property name="geometry">
     <rect>
      <x>175</x>
      <y>59</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lru_total_time_label">
    <property name="geometry">
     <rect>
      <x>417</x>
      <y>59</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lfu_total_time_label">
    <property name="geometry">
     <rect>
      <x>676</x>
      <y>59</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="opt_total_time_label">
    <property name="geometry">
     <rect>
      <x>947</x>
      <y>59</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lru_ave_time_label">
    <property name="geometry">
     <rect>
      <x>417</x>
      <y>105</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="fifo_ave_time_label">
    <property name="geometry">
     <rect>
      <x>175</x>
      <y>105</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lfu_ave_time_label">
    <property name="geometry">
     <rect>
      <x>676</x>
      <y>105</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="opt_ave_time_label">
    <property name="geometry">
     <rect>
      <x>947</x>
      <y>105</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="fifo_miss_page_label">
    <property name="geometry">
     <rect>
      <x>175</x>
      <y>151</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lru_miss_page_label">
    <property name="geometry">
     <rect>
      <x>417</x>
      <y>151</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="lfu_miss_page_label">
    <property name="geometry">
     <rect>
      <x>676</x>
      <y>151</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="opt_miss_page_label">
    <property name="geometry">
     <rect>
      <x>947</x>
      <y>151</y>
      <width>81</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="widget_4" native="true">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>260</y>
     <width>1061</width>
     <height>251</height>
    </rect>
   </property>
   <widget class="QTextEdit" name="fifo_textEdit">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>10</y>
      <width>201</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="lru_textEdit">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>10</y>
      <width>211</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="lfu_textEdit">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>10</y>
      <width>211</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextEdit" name="opt_textEdit">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>10</y>
      <width>211</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QWidget" name="widget_3" native="true">
   <property name="geometry">
    <rect>
     <x>90</x>
     <y>217</y>
     <width>1051</width>
     <height>41</height>
    </rect>
   </property>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>10</y>
      <width>41</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>FIFO</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>10</y>
      <width>41</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>LRU</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>10</y>
      <width>41</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>LFU</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>890</x>
      <y>10</y>
      <width>41</width>
      <height>18</height>
     </rect>
    </property>
    <property name="text">
     <string>OPT</string>
    </property>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
